components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -4637,24 +4662,31 @@ public int getApplicationsCount(int tenantId, String searchOwner, String searchA
      * Returns all the consumerkeys of application which are subscribed for the given api
      *
      * @param identifier APIIdentifier
+     * @param organization Organization
      * @return Consumerkeys
      * @throws org.wso2.carbon.apimgt.api.APIManagementException if failed to get Applications for given subscriber.
      */
-    public String[] getConsumerKeys(APIIdentifier identifier) throws APIManagementException {
+    public String[] getConsumerKeys(APIIdentifier identifier, String organization) throws APIManagementException {
 
         Set<String> consumerKeys = new HashSet<String>();
 
         Connection connection = null;
         PreparedStatement prepStmt = null;
         ResultSet rs = null;
 
+        String uuid;
+        if (identifier.getUUID() != null) {
+            uuid = identifier.getUUID();
+        } else {
+            uuid = getUUIDFromIdentifier(identifier, organization);
+        }
```

Comment:
these kind of branching can cause issues..
Shall we avoid passing APIIdentifier in the first place and pass uuid instead?

Reply:
@tharikaGitHub seems like there are no usages of getConsumerKeys() method.

Reply:
Will change this.

Reply:
This method is removed with https://github.com/wso2/carbon-apimgt/pull/10587